amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/MultiStreamLease.java

```diff
@@ -0,0 +1,91 @@
+package software.amazon.kinesis.leases;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang3.Validate;
+
+import java.util.Objects;
+
+import static com.google.common.base.Verify.verifyNotNull;
+
+@Setter
+@NoArgsConstructor
+@Getter
+@Accessors(fluent = true)
+public class MultiStreamLease extends Lease {
+
+    @NonNull private String streamName;
+    @NonNull private String shardId;
+
+    public MultiStreamLease(Lease other) {
```

Comment:
What's the goal of this object? Seems like its same as lease object. Allowing both Lease and MultiStreamLease to be used within the same app will be confusing. If the app initialized to use multi-stream-leases, then it should be initializing multi-stream leases, not initializing leases and casting to multi-stream lease. 

If the only difference between the two is serialization, we should see if the multi-stream-lease abstraction should be added at the lease level, or at the higher serialization level according to the application configuration.

Reply:
We either use Lease or Multistream lease in an application. This is determined through the serializer we set while initializing the scheduler. MultiStreamLease supports serializing and deserializing new fields like streamName and shardId, which we dont support with current Lease. We can rename this to be LeaseV2, as we might add pendingCheckpointState as well to for two phase commit.